tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -592,7 +688,11 @@ export class WebGPUBackend extends KernelBackend {
   }
 
   private binaryCompareOp(a: Tensor, b: Tensor, op: string): Tensor {
-    const program = new BinaryOpProgram(op, a.shape, b.shape);
+    const key = this.getHash([binary_op.getOpNum(op), a.shape, b.shape,
```

Comment:
Can you let binaryCompareOp to reuse `binaryOp` so that we only need to add key in `binaryOp`? Maybe change the prototype of binaryOp to `binaryOp(a: Tensor, b: Tensor, op: string, compareOp?:bool)`